internal/serialization/serialization.go

```diff
@@ -83,16 +86,20 @@ func (s *Service) ToObject(data *Data) (r interface{}, err error) {
 			err = makeError(rec)
 		}
 	}()
+	var ok bool
 	if data == nil {
 		return nil, nil
 	}
 	typeID := data.Type()
 	if typeID == 0 {
 		return data, nil
 	}
```

Comment:
Is this correct? If the `typeId == 0` (I would also prefer using some constant there instead of magic number of 0), then the data contains no payload, hence represents `nil`. But, we are not returning `nil` here but return the data instead?

Reply:
That doesn't seem correct to me, removed that check: https://github.com/hazelcast/hazelcast-go-client/pull/581/commits/173bfd382d8de457c1bf1f3cd13b7dfb9d9dcb45